controllers/buildpipeline/buildpipeline_controller.go

```diff
@@ -92,17 +92,21 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 				return ctrl.Result{}, err
 			}
 		}
-		return helpers.HandleLoaderError(logger, err, "Component", "PipelineRun")
+		logger.Error(err, "Failed to get component for build pipeline", "name", pipelineRun.Name, "namespace", pipelineRun.Namespace)
+		return ctrl.Result{}, nil
```

Comment:
Should we `return ctrl.Result{}, err` here, then reconcile the build PLR again later?

Reply:
We should.  I think on linke 92 we should be returning `ctrl.Result{}, nil` too. I don't think we want to re-reconcile if the component can't be found.  Do you agree?

Reply:
If we return `ctrl.Result{}, nil` on line 92, wouldn't we have a build PLR, with a finalizer attached to it, left hanging in the cluster?

Reply:
> If we return `ctrl.Result{}, nil` on line 92, wouldn't we have a build PLR, with a finalizer attached to it, left hanging in the cluster?

Agreed, on line 92 an error happened while trying to remove the finalizer, so our reconciliation didn't do everything it needed to do, hence the error being returned to requeue a subsequent reconciliation attempt.

Reply:
Tbh I'm in favor in leaving the `return helpers.HandleLoaderError(logger, err, "Component", "PipelineRun")` call here instead of making the current change, because that will handle the case when the component is not found as well as when there's a different error finding it.

Reply:
Right we do handle the removal of the finalizer above currently, but not the logging and returning `ctrl.Result{}, nil`.

We could add another `return ctrl.Result{}, nil` to line 93 to stop reconciliation and prevent requeuing when the component is noth there, but I don't think it's worth adding when `HandleLoaderError` would do the same thing.

Reply:
I see what you mean.  I've added back the call to `HandleLoaderError`